src/backend/api/meals.js

```diff
@@ -1,12 +1,193 @@
-const express = require("express");
+const express = require('express');
 const router = express.Router();
-const knex = require("../database");
+const knex = require('../database');
+let errorTitle = '';
+let q = '';
+function mealsQueryBuilder(query) {
+  try {
+    errorTitle = '';
+    if (query.hasOwnProperty('availableReservations')) {
+      q = knex
+        .select(
+          'Meal.title',
+          knex.raw('max_reservations - number_of_guest as ??', [
+            'Available_reservation',
+          ])
+        )
+        .from('Meal')
+        .join(
+          knex('Reservation')
+            .select(
+              'meal_id',
+              knex.raw('sum(number_of_guests) as ??', ['number_of_guest'])
+            )
+            .groupBy('meal_id')
+            .as('guestNumber'),
+          'id',
+          'guestNumber.meal_id'
+        );
+      q = q.whereRaw(
+        `max_reservations - number_of_guest ${
+          JSON.parse(query.availableReservations) ? '>' : '<='
+        } 0`
+      );
+    }
+    if (query.hasOwnProperty('maxPrice')) {
+      const maxPrice = parseInt(query.maxPrice);
+      if (isNaN(maxPrice)) {
+        errorTitle = 'maxPrice';
+      } else {
+        const cheapMeals = knex('Meal').where('price', '<', maxPrice);
+        q = cheapMeals;
```

Comment:
Oh q is not query, its the result or? 

Remember good variable names. Here it is actually confusing